axelrod/result_set.py

```diff
@@ -13,132 +14,555 @@
 class ResultSet(object):
     """A class to hold the results of a tournament."""
 
-    def __init__(self, players, turns, repetitions, outcome,
-                 with_morality=True):
+    def __init__(self, players, matches, with_morality=True):
         """
         Args:
             players (list): a list of player objects.
-            turns (int): the number of turns per interaction.
-            repetitions (int): the number of time the round robin was repeated.
-            outcome (dict): returned from the Tournament class and containing
-                various sets of results for processing by this class.
+            matches (list): a list of list of completed matches
+                            (1 for each repetition)
             with_morality (bool): a flag to determine whether morality metrics
                 should be calculated.
         """
```

Comment:
This whole docstring needs to be in numpy format as we agreed in #347 

Whilst we've not bothered going through existing docstrings and changing them in one go for that ticket, we have been doing so whenever they've been 'touched' as part of other changes.